opal/corelib/number.rb

```diff
@@ -710,6 +710,10 @@ def positive?
   def negative?
     `self == -Infinity || 1 / self < 0`
   end
+
+  def instance_variable_set(*)
+    raise RuntimeError, "can't modify frozen #{self.class.name}"
```

Comment:
@meh to me looks like it should fail anyway, looks odd to store ivars on a numeric object (even if I can confirm it works on latest safari) no strong feelings tho